src/test/java/mate/academy/springbootintro/controllertests/BookControllerTest.java

```diff
@@ -0,0 +1,290 @@
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get all books")
+    public void getAll_FindAllBooks_Correct() throws Exception {
+        BookDto book1 = new BookDto();
+        book1.setId(1L);
+        book1.setTitle("Harry Potter");
+        book1.setAuthor("Joanne Rowling");
+        book1.setIsbn("12345");
+        book1.setPrice(BigDecimal.valueOf(50));
+        book1.setDescription("Book about boy...");
+        book1.setCoverImage("img.harry.potter");
+
+        BookDto book2 = new BookDto();
+        book2.setId(2L);
+        book2.setTitle("The little prince");
+        book2.setAuthor("Antoine de Saint-Exupery");
+        book2.setIsbn("4557839");
+        book2.setPrice(BigDecimal.valueOf(80));
+        book2.setDescription("Story about little prince");
+        book2.setCoverImage("img.little.prince");
+
+        BookDto book3 = new BookDto();
+        book3.setId(1L);
+        book3.setTitle("The hobbit");
+        book3.setAuthor("J. R. R. Tolkien");
+        book3.setIsbn("93587");
+        book3.setPrice(BigDecimal.valueOf(40));
+        book3.setDescription("Book about hobbit Bilbo");
+        book3.setCoverImage("img.hobbit");
+
+        List<BookDto> excepted = List.of(book1, book2, book3);
+
+        MvcResult result = mockMvc.perform(
+                get("/books")
+                        .contentType(MediaType.APPLICATION_JSON)
+        )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto[] resultBooks = objectMapper.readValue(
+                result.getResponse().getContentAsString(), BookDto[].class);
+        List<BookDto> actual = Arrays.stream(resultBooks).toList();
+        assertEquals(3, actual.size());
+        assertEquals(excepted, actual);
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get book by id")
+    public void getBookById_FindBookWithValidId_Correct() throws Exception {
+        BookDto excepted = new BookDto();
+        excepted.setId(1L);
+        excepted.setTitle("Harry Potter");
+        excepted.setAuthor("Joanne Rowling");
+        excepted.setIsbn("12345");
+        excepted.setPrice(BigDecimal.valueOf(50));
+        excepted.setDescription("Book about boy...");
+        excepted.setCoverImage("img.harry.potter");
+
+        MvcResult result = mockMvc.perform(get("/books/1")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), BookDto.class);
+
+        assertNotNull(actual);
+        assertEquals(excepted.getId(), actual.getId());
+        assertEquals(excepted.getTitle(), actual.getTitle());
+        assertEquals(excepted.getAuthor(), actual.getAuthor());
+        assertEquals(excepted.getIsbn(), actual.getIsbn());
```

Comment:
instead of such serial comparisons make book override equals, and use one assert